util/data_set_helpers.py

```diff
@@ -35,6 +39,80 @@ def dev(self):
+        self.example_queue = tf.PaddingFIFOQueue(shapes=[[None, numcep + (2 * numcep * numcontext)], [], [None,], []],
+                                                  dtypes=[tf.float32, tf.int32, tf.int32, tf.int32],
+                                                  capacity=2 * self._get_device_count() * batch_size)
+        self._enqueue_op = self.example_queue.enqueue([self._x, self._x_length, self._y, self._y_length])
+        self._close_op = self.example_queue.close(cancel_pending_enqueues=True)
+        self._filelist = filelist
+        self.batch_size = batch_size
+        self._numcontext = numcontext
+        self._thread_count = thread_count
+        self._files_list = self._create_files_list()
+        self._next_index = next_index
+
+    def _get_device_count(self):
+        available_gpus = get_available_gpus()
+        return max(len(available_gpus), 1)
+
+    def start_queue_threads(self, session, coord):
+        self._coord = coord
+        batch_threads = [Thread(target=self._populate_batch_queue, args=(session,)) for i in range(self._thread_count)]
+        for batch_thread in batch_threads:
+            batch_thread.daemon = True
+            batch_thread.start()
+        return batch_threads
+
+    def close_queue(self, session):
+        session.run(self._close_op)
+
+    def _create_files_list(self):
+        # 1. Sort by wav filesize
+        # 2. Select just wav filename and transcript columns
+        # 3. Return a NumPy representation
+        return self._filelist.sort_values(by="wav_filesize")        \
+                             .ix[:, ["wav_filename", "transcript"]] \
+                             .values
+
+    def _indices(self):
+        index = -1
+        while not self._coord.should_stop():
+            index = self._next_index(index) % len(self._files_list)
+            yield self._files_list[index]
+
+    def _populate_batch_queue(self, session):
+        for wav_file, transcript in self._indices():
+            source = audiofile_to_input_vector(wav_file, self._numcep, self._numcontext)
+            source_len = len(source)
+            target = text_to_char_array(transcript)
+            target_len = len(target)
+            try:
+                session.run(self._enqueue_op, feed_dict={
+                    self._x: source,
+                    self._x_length: source_len,
+                    self._y: target,
+                    self._y_length: target_len})
+            except tf.errors.CancelledError:
+                return
+
+    def next_batch(self):
+        source, source_lengths, target, target_lengths = self.example_queue.dequeue_many(self.batch_size)
+        sparse_labels = ctc_label_dense_to_sparse(target, target_lengths, self.batch_size)
+        return source, source_lengths, sparse_labels
+
+    @property
+    def total_batches(self):
+        # Note: If len(_filelist) % batch_size != 0, this re-uses initial files
+        return int(ceil(len(self._filelist) / self.batch_size))
```

Comment:
The import
```python
from __future__ import absolute_import, division, print_function
```
changes the value returned here.

For example if `len(self._filelist)` were 10 and `self.batch_size` were 6, then in python 2.x one finds
```bash
>>> from math import ceil
>>> int(ceil(10 / 6))
1
>>> from __future__ import division
>>> int(ceil(10 / 6))
2
```
So this changes the `total_batches` property.